packages/markers/src/browser/marker-manager.ts

```diff
@@ -39,13 +39,12 @@ export class MarkerCollection<T> {
     }
 
     setMarkers(owner: string, markerData: T[]): Marker<T>[] {
-        const before = this.owner2Markers.get(owner);
```

Comment:
Could you elaborate why the contract should be changed? It used to return old markers.

Reply:
The function seems to be the same, `before` wasn't used except in the `return`, which was changed:
```ts
return before || []; // before
return this.owner2Markers.get(owner) || []; // after
```

Reply:
But it used to return the old markers, and now it will return the new markers.

Reply:
Oh right m bad

Reply:
@marechal-p I think it's preferrable not to remove your message, it just makes the discussion more difficult to read.

Reply:
so it should be reverted since it changes the semantic for no reason?

Reply:
> so it should be reverted since it changes the semantic for no reason?

Yes.

Reply:
Keeping the old markers does not make sence when you have a new set of markers for this file. When you make a modification to remove the problems, you don't want to have a list of obsolete markers being maintained, you want the latest information about this file. So with the change, wether you have a new set of markers or no more markers, you want to update the list.

Reply:
The list of old markers that is returned is not kept anywhere, it's just returned to the caller, if they want to do something with this information.  But in the marker manager, we always keep the latest info, regardless.

Actually, no caller seem to use the return value anyway, so maybe we could just remove the return value?

Reply:
> Actually, no caller seem to use the return value anyway, so maybe we could just remove the return value?

True, as far as we know. But an "outside" extender might be using the marker-manager and depending on the old behavior. So I think the default approach of having to reasonably justify changing a public API, is proper.

Reply:
The return value is indirectly used in a test (`problem-manager.spec.ts`), and it is only by luck that the test still passes: The test check the number of errors before / after, turns out it is 2 for both cases.

Anyway this modification should be reverted as it adds nothing. If we want the new markers we call `collection.getMarkers()` after the `collection.setMarkers()` (especially in `MarkerManager.internalSetMarkers`).

Reply:
I put back the line in setMarkers() to return the old list of markers.
Question: On a set() method we should have two choices: either no return value since we are setting, adding,  removing a value  or we should return the result of the setting, not returning what was available before we call set() method. When a user is looking of what was available before a set() method, he can perform a get() and use that data as he wishes. We should not depend of the return of a set() to GET the OLD data. I think this is a bad use of a set() method

Reply:
> The return value is indirectly used in a test (problem-manager.spec.ts), and it is only by luck that the test still passes: The test check the number of errors before / after, turns out it is 2 for both cases.

Ahh you're right I missed that.  Well, maybe this test should be improved then :)

> Anyway this modification should be reverted as it adds nothing. If we want the new markers we call collection.getMarkers() after the collection.setMarkers() (especially in MarkerManager.internalSetMarkers).

Why would we want the new markers in internalSetMarkers, isn't it supposed to return the old ones?

Reply:
> Why would we want the new markers in internalSetMarkers, isn't it supposed to return the old ones?

Yes but I think @lmcbout added a check to delete an entry as [part of an optimization I think](https://github.com/theia-ide/theia/pull/2093/files#diff-fa0fb0be865e7c3cf2af196391b00cf6R187).